source/retrieve-data.txt

```diff
@@ -0,0 +1,196 @@
+   :manual:`natural order </reference/glossary/#std-term-natural-order>`
+   on disk if no sort criteria is specified.
+
+.. TODO: uncomment when sort guide made
+.. To learn more about sorting, see the :ref:`sort guide <java-rs-sort>`.
+
+.. _java-rs-retrieve-find-multiple:
+
+Find Multiple Documents
+~~~~~~~~~~~~~~~~~~~~~~~
+
+To find multiple documents in a collection, pass a query filter to the ``find()``
+method that specifies the criteria of the documents you want to retrieve.
+
+The following example uses the ``find()`` method to find all documents where
+the ``"cuisine"`` field has the value ``"Spanish"``:
+
+.. code-block:: java
+   :copyable: true
+
+   FindPublisher<Document> findDocPublisher = restaurants.find(
+           eq("cuisine", "Spanish"));
+   List<Document> findResult = Flux.from(findDocPublisher).collectList().block();
+
+You can iterate over the documents you retrieved using the ``find()`` method by using a ``for`` loop, as shown in
+the following example:
+
+.. code-block:: java
+   :copyable: true
+
+   FindPublisher<Document> findDocPublisher = restaurants.find(
+           eq("cuisine", "Spanish"));
+   List<Document> findResult = Flux.from(findDocPublisher).collectList().block();
+   
+   for (Document document : findResult) {
+   ...
+   }
+
+.. note:: Find All Documents
+
+   To find all documents in a collection, pass no parameters to the ``find()`` method:
+
+   .. code-block:: java
+
+      Publisher<Document> findAllPublisher = restaurants.find();
+      List<Document> findResult =
+              Flux.from(findAllPublisher).collectList().block();
+      
+
+Modify Find Behavior
+~~~~~~~~~~~~~~~~~~~~
+
+You can modify the behavior of the ``find()`` method by chaining methods
+to it. The following table describes commonly used methods:
+
+.. list-table::
+   :widths: 30 70
+   :header-rows: 1
+
+   * - Argument
+     - Description
+
+   * - ``batchSize(int batchSize)`` 
+     - | Limits the number of documents to hold in a cursor at a given time.
+
+   * - ``collation(Collation collation)`` 
+     - | Sets the collation options as an instance of the Collation class. 
+
+   * - ``comment(String comment)`` 
+     - | Attaches a string to the query. This can help you trace and interpret the
+         operation in the server logs and in profile data. To learn more about query comments, 
+         see the :manual:`$comment </reference/operator/query/comment/>` page.
+   
+   * - ``hint(Bson hint)`` 
+     - | Sets an index to use for the query.
+
+   * - ``maxTime(long maxTime, TimeUnit timeUnit)`` 
+     - | Sets the maximum execution time on the server for this operation. If this time is
+         exceeded, the {+driver-short+} aborts the operation and raises an ``ExecutionTimeout``.
+
+The following example uses the ``find()`` method to find all documents where
+the ``"cuisine"`` field has the value ``"Italian"`` and sets a maximum execution
+time of 10 seconds (10,000 milliseconds):
```

Comment:
S: The seconds/milliseconds clarification is great in most cases, but the example explicitly uses ``TimeUnit.SECONDS``, so I don't think it's needed

```suggestion
time of 10 seconds:
```